src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/EventProvider.cs

```diff
@@ -471,7 +471,7 @@ private unsafe void GetSessionInfo(Action<int, long> action)
 
             // Determine our session from what is in the registry.  
             string regKey = @"\Microsoft\Windows\CurrentVersion\Winevt\Publishers\{" + m_providerId + "}";
-            if (System.Runtime.InteropServices.Marshal.SizeOf(typeof(IntPtr)) == 8)
+            if (sizeof(IntPtr) == 8)
```

Comment:
I have initially converted everything to sizeof(IntPtr) but then realized that sizeof(IntPtr) works in unsafe context only. I have converted the places that broke to IntPtr.Size so that I do not have to add the unsafe contexts.